build/build-sdk-images/csharp/gen.sh

```diff
@@ -16,13 +16,52 @@
 
 set -ex
 
+header() {
+    cat /go/src/agones.dev/agones/build/boilerplate.go.txt "$1" | sponge "$1"
+}
+
 sdk=/go/src/agones.dev/agones/proto/sdk
 googleapis=/go/src/agones.dev/agones/proto/googleapis
+protoc_intermediate=/go/src/agones.dev/agones/sdks/csharp/proto
 protoc_destination=/go/src/agones.dev/agones/sdks/csharp/sdk/generated
 
-# Generate C# proto file `Sdk.cs`
-protoc --csharp_out=${protoc_destination} -I ${sdk} -I ${googleapis} --plugin=protoc-gen-grpc=`which grpc_csharp_plugin` sdk.proto
-protoc --csharp_out=${protoc_destination} -I ${sdk}/alpha -I ${googleapis} --plugin=protoc-gen-grpc=`which grpc_csharp_plugin` alpha.proto
-# Generate proto stub?
-protoc --grpc_out=${protoc_destination} -I ${sdk} -I ${googleapis} --plugin=protoc-gen-grpc=`which grpc_csharp_plugin` sdk.proto
-protoc --grpc_out=${protoc_destination} -I ${sdk}/alpha -I ${googleapis} --plugin=protoc-gen-grpc=`which grpc_csharp_plugin` alpha.proto
+# Create temporary proto files
+mkdir -p ${protoc_intermediate}
+cp -r ${sdk} ${protoc_intermediate}
+
+# Remove protoc-gen-openapiv2 definitions because C# package doesn't support grpc-gateway
+sed -i -e 's/import "protoc-gen-openapiv2\/options\/annotations.proto";//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/info: {//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/title: "sdk.proto";//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -z 's/version: "version not set";\n    };//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/schemes: HTTP;//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/consumes: "application\/json";//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -z 's/produces: "application\/json";\n};//' ${protoc_intermediate}/sdk/sdk.proto
+sed -i -e 's/bool disabled = 1.*/bool disabled = 1;/' ${protoc_intermediate}/sdk/sdk.proto
+
+sed -i -e 's/import "protoc-gen-openapiv2\/options\/annotations.proto";//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/info: {//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/title: "alpha.proto";//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -z 's/version: "version not set";\n    };//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/schemes: HTTP;//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/consumes: "application\/json";//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -z 's/produces: "application\/json";\n};//' ${protoc_intermediate}/sdk/alpha/alpha.proto
+sed -i -e 's/bool bool = 1.*/bool bool = 1;/' ${protoc_intermediate}/sdk/alpha/alpha.proto
```

Comment:
This is workaround. Remove protoc-gen-openapiv2 definitions from C# sdk image because C# package doesn't support grpc-gateway